interfaces/HO-Portal/src/app/programs/program-details/program-details.component.ts

```diff
@@ -1,32 +1,108 @@
 import { Component, OnInit } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
 import { ProgramsServiceApiService } from 'src/app/services/programs-service-api.service';
 import { Program } from 'src/app/models/program.model';
+import { ModalController } from '@ionic/angular';
+import { ProgramModalPage } from '../program-modal/program-modal.page';
+import { OverlayEventDetail } from '@ionic/core';
+import { TranslateService } from '@ngx-translate/core';
 
 @Component({
   selector: 'app-program-details',
   templateUrl: './program-details.component.html',
   styleUrls: ['./program-details.component.scss'],
 })
 export class ProgramDetailsComponent implements OnInit {
-  public program: any;
+  public languageCode: string;
+  public fallbackLanguageCode: string;
+
+  public program: Program;
+  public programTitle: string;
+  public programArray: any;
+
+  private techFeatures = [
+    'countryId',
+    'schemaId',
+    'credDefId',
+    'credOffer',
+    'proofRequest',
+  ];
+
 
   constructor(
     private route: ActivatedRoute,
-    private router: Router,
-    private programsService: ProgramsServiceApiService
-  ) { }
+    private programsService: ProgramsServiceApiService,
+    public modalController: ModalController,
+    public translate: TranslateService,
+  ) {
+
+  }
 
   ngOnInit() {
+    this.fallbackLanguageCode = this.translate.getDefaultLang();
+    this.languageCode = this.translate.currentLang;
+
     const id = this.route.snapshot.paramMap.get('id');
     this.programsService.getProgramById(id).subscribe((response) => {
-      this.program = this.generateArray(response);
-      console.log(this.program);
+      this.program = response;
+      this.programTitle = this.mapLabelByLanguageCode(this.program.title);
+      this.programArray = this.generateArray(this.program);
     });
   }
 
+
+
   public generateArray(obj) {
-    return Object.keys(obj).map((key) => ({ key, value: obj[key] }));
+    const result = [];
+    for (const key in obj) {
+      if (this.techFeatures.indexOf(key) <= -1) {
```

Comment:
I think using one of the 'newer' array-functions, like `.map()` or `.filter()` (See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#Iteration_methods) would make it easier to keep this method small..

Reply:
adjusted